test/fixtures.py

```diff
@@ -0,0 +1,51 @@
+"""This file contains all fixtures that are needed by multiple test files."""
+from shlex import split
+from subprocess import run
+from textwrap import dedent
+
+from pytest import fixture
+
+from docker_cli.utils import generate_codename
+
+from .utils import determine_scope, remove_docker_image
+
+
+@fixture(scope=determine_scope)
+def image_tag():
+    """
+    Returns an image tag
+
+    Returns
+    ------
+    str
+        An image tag
+    """
+    tag = f"isce3_pytest_temp_{generate_codename(k=10)}"
+    yield tag
+
+
+@fixture(scope="function")
+def image_id(image_tag):
+    """
+    Builds an image for testing and returns its ID.
+
+    Yields
+    ------
+    str
+        An image ID.
+    """
+    dockerfile = dedent(f"""\
+        FROM ubuntu
+
+        RUN echo {image_tag}
+    """.rstrip())
```

Comment:
Instead of escaping the initial newline character and stripping the last newline character, you could just strip both at once -- but you have to `dedent()` before you `strip()`!

```suggestion
    dockerfile = dedent(f"""
        FROM ubuntu

        RUN echo {image_tag}
    """).strip()
```